### PR TITLE
UI – Adding in Permission component places

### PR DESCRIPTION
**Pull request by jcontonio**
* Updates projectId type on ProjectPermission component
* Adding in permissions check to Navigator, ProjectListItem, and ProjectSettings

### CODE REVIEW

**Comment by jjidt** [comment]
@jcontonio pushing this back to you -

- fixed loading state issues with user-assignments and projects
- sets an internal mounted flag in useDataService to prevent setting internal state if the component is unmounted

File: web/src/client/js/libs/makeCancelable.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+const makeCancelable = (promise) => {
+  let hasCanceled_ = false
```
Comment (jcontonio) [inline web/src/client/js/libs/makeCancelable.js]: what's this underscore indicate?

Comment (jjidt) [reply web/src/client/js/libs/makeCancelable.js]: wait, didn't I delete this file?

Comment (jjidt) [reply web/src/client/js/libs/makeCancelable.js]: that was copied code from a blog post

Comment (jcontonio) [reply web/src/client/js/libs/makeCancelable.js]: Oh, yeah you did in the following commits.

**Review by jjidt — APPROVED** [review]

**Pull request merged**